animating-widgets.md

```diff
@@ -0,0 +1,293 @@
+objects’s value and status changes over
+time) as it runs.
+
+Based on the animation object’s properties at a specific frame, Flutter
+dynamically modifies your widget visual appearence, and rebuilds the widget
+tree to produce visual transitions such as sliding, scaling, rotation, height
+or width changes, and fading.
+
+Flutter's flexible animation model allows you to define interpolations for
+spatial positioning, color gradients, image scaling, and others. You can define
+your own subclass of
+[`Animation`](https://docs.flutter.io/flutter/animation/Animation-class.html)
+or use an animation subclass that the framework provides.
+
+To learn more about
+using the other key classes in the
+[`flutter/animation`](https://docs.flutter.io/flutter/animation/animation-library.html)
+package, see the [Animations in Flutter](/animations) guide.
+
+## Render a basic widget animation
+
+To render your widget with an animation, set one or more of the widget’s
+members to the animation object, then use the animation’s value to decide how
+to draw your object.
+
+If you are animating a stateful widget, you can notify your widget of changes
+to the animation’s values by using the
+[`addListener()`](https://docs.flutter.io/flutter/animation/Animation/addListener.html)
+method. The framework invokes the listener callback whenever the value of the
+animation changes. Typically, you’ll want to call the
+[`setState()`](http://docs.flutter.io/flutter/material/State/setState.html)
+method when the animation value change occurs, to trigger the framework to
+rebuild the widget tree.
+
+The following code snippet shows how you might create a simple animation
+effect to make an logo image appear to expand.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter/animation.dart';
+
+class LogoApp extends StatefulWidget {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> {
+  Animation<double> animation;
+
+  initState() {
+    AnimationController controller = new AnimationController(
+      duration: const Duration(milliseconds: 5000)
+    );
+    animation = new Tween(begin: 0.0, end: 200.0).animate(controller);
+    animation.addListener(() {
+      setState(() {
+      // the state that has changed here is the animation object’s value
+      });
+    });
+    controller.forward();
+    super.initState();
+  }
+
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: animation.value,
+        width: animation.value,
+        child: new AssetImage(
+          name: 'assets/flutter_logo.png',
+          fit: ImageFit.contain
+        )
+      )
+    );
+  }
+}
+
+void main() {
+  runApp(
+    new MaterialApp(
+      title: 'My Animation Test',
+      routes: <String, WidgetBuilder> {
+        '/': (BuildContext context) => new LogoApp()
+      }
+    )
```

Comment:
Why were all of the "new MaterialApp" statements added to the examples? These are not Material Apps and so similar statements were removed in early code reviews of these examples.